lib/airbrake/logger/logger_ext.rb

```diff
@@ -0,0 +1,78 @@
+##
+# Redefine +Logger+ from stdlib, so it can both log and report errors to
+# Airbrake.
+#
+# @example
+#   # Create a logger like you normally do.
+#   logger = Logger.new(STDOUT)
+#
+#   # Assign a default Airbrake notifier
+#   logger.airbrake = Airbrake[:default]
+#
+#   # Just use the logger like you normally do.
+#   logger.fatal('oops')
+class Logger
+  # Store the orginal methods to use them later.
+  alias add_without_airbrake add
+  alias initialize_without_airbrake initialize
+
+  ##
+  # @see https://goo.gl/MvlYq3 Logger#initialize
+  def initialize(*args)
+    @airbrake = Airbrake[:default]
+    @airbrake_severity_level = WARN
+    initialize_without_airbrake(*args)
+  end
+
+  ##
+  # @return [Airbrake::Notifier] notifier to be used to send notices
+  attr_accessor :airbrake
+
+  ##
+  # @example
+  #   logger.airbrake_severity_level = Logger::FATAL
+  # @return [Integer] the level that c
+  attr_accessor :airbrake_severity_level
+
+  ##
+  # @see https://goo.gl/8zPyoM Logger#add
+  def add(severity, message = nil, progname = nil, &block)
+    if severity >= airbrake_severity_level && airbrake
+      notify_airbrake(severity, message || progname)
+    end
+    add_without_airbrake(severity, message, progname, &block)
+  end
+
+  private
+
+  def notify_airbrake(severity, message)
+    notice = Airbrake.build_notice(message)
+
+    # Get rid of unwanted internal Logger frames. Examples:
+    # * /ruby-2.4.0/lib/ruby/2.4.0/logger.rb
+    # * /gems/activesupport-4.2.7.1/lib/active_support/logger.rb
+    backtrace = notice[:errors].first[:backtrace]
+    notice[:errors].first[:backtrace] =
+      backtrace.drop_while { |frame| frame[:file] =~ %r{/logger.rb\z} }
+
+    notice[:context][:component] = 'log'
+    notice[:context][:severity] = airbrake_severity(severity)
+
+    Airbrake.notify(notice)
```

Comment:
Uh, of course, it's a mistake indeed. What about `logger.airbrake_notifier`? We monkey-patch a class and it's better to be verbose.